src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -314,13 +351,6 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
                     }
                 }
 
-                ODataEntityReferenceLinkBase entityReferenceLink = childItem as ODataEntityReferenceLinkBase;
-                if (entityReferenceLink != null)
-                {
-                    // ignore entity reference links.
-                    continue;
-                }
-
```

Comment:
Curious to know why the logic of the reference links happens before the loop instead of at this point?